### PR TITLE
Publish status event on bind for SMPP transport

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: vumi/transports/smpp/smpp_transport.py [hunk 1]
```diff
@@ -299,6 +299,16 @@ def _reject_for_invalid_address(self, message, field):
...
 
     @inlineCallbacks
+    def on_smpp_bind(self):
+        yield self.publish_status_bound()
+
+    def publish_status_bound(self):
+        return self.publish_status('smpp', 'good', reasons=[{
+            'type': 'bound',
+            'message': 'Bound',
+        }])
```
Comment (justinvdm) [inline vumi/transports/smpp/smpp_transport.py]: Ideas on any `details` we should add here?

Comment (rudigiesler) [inline vumi/transports/smpp/smpp_transport.py]: I don't think we have any information at this point that would be useful? I think things like the time it took to bind, or smpp version. But I don't think that any of those things are easily available here, we would have to implement them separately. I think for now we can leave details empty, and if we find that some things are useful to know, then we can implement them.

**Comment by rudigiesler** [comment]
Do you think we should be adding the relevant documentation with each of these PRs, or just do one big PR in the end with all of these changes?

File: vumi/transports/smpp/smpp_transport.py (hunk 1 above)
Comment (justinvdm) [inline vumi/transports/smpp/smpp_transport.py]: +1

**Comment by justinvdm** [comment]
> Do you think we should be adding the relevant documentation with each of these PRs, or just do one big PR in the end with all of these changes?

Was thinking about this, decided to go for big docs PR at the end of all these changes to avoid needing to keep the docs up to date on each small change we make to these first few events.

**Comment by rudigiesler** [comment]
:+1:

**Pull request merged**